### PR TITLE
taskrunner/many: KnownTaskKinds helper

### PR DESCRIPTION
**Pull request by stolowski**
Added KnowTaskKinds helper to taskrunner and all managers, in preparation for implementing handling of unknown tasks.

### CODE REVIEW

File: overlord/state/taskrunner.go [hunk 1]
```diff
@@ -84,6 +84,15 @@ func (r *TaskRunner) AddHandler(kind string, do, undo HandlerFunc) {
 	r.handlers[kind] = handlerPair{do, undo}
 }
 
+// KnownTaskKinds returns all tasks kinds handled by this runner.
+func (r *TaskRunner) KnownTaskKinds() []string {
+	kinds := make([]string, 0, len(r.handlers))
+	for h := range r.handlers {
```
Comment (bboozzoo) [inline overlord/state/taskrunner.go]: Is it possible to have more than one task of given kind?

Comment (stolowski) [reply overlord/state/taskrunner.go]: No, this is not about tasks, it's about their kinds (= names of handlers). They are unique.

File: overlord/cmdstate/cmdmgr.go [hunk 2]
```diff
@@ -41,6 +41,10 @@ func Manager(st *state.State) *CommandManager {
 	return &CommandManager{runner: runner}
 }
 
+func (m *CommandManager) KnownTaskKinds() []string {
+	return m.runner.KnownTaskKinds()
```
Comment (niemeyer) [inline overlord/cmdstate/cmdmgr.go]: Can we please have a test for each of those, for the real ones (not mock/fake)?

It sounds silly, but a single one of those that happens to return []string{} will end up with a hard-to-catch bug, where tasks will magically disappear in real deployments.

Let's please double check that all the real ones are indeed returning the right thing.

Comment (niemeyer) [inline overlord/cmdstate/cmdmgr.go]: Can we please have a test for each of those, for the real ones (not mock/fake)?

It sounds silly, but a single one of those that happens to return []string{} will end up with a hard-to-catch bug, where tasks will magically disappear in real deployments.

Let's please double check that all the real ones are indeed returning the right thing.

**Comment by stolowski** [comment]
Marking as blocked for now so that it's not accidentally landed before I add the requested unit tests.

**Pull request merged**